### ISSUE TITLE
[Bug]: Transfer ends up with two entries on different dates

### ISSUE THREAD

**Issue by achennu** [actualbudget/actual#1601]
### Verified issue does not already exist?

- [X] I have searched and found no existing issue

### What happened?

I have two accounts, `A` and `B`, both of which are setup to sync bank accounts automatically via gocardless.

I made a transfer of N euros from `A` to `B` on August 4th. The funds arrived in account `B` on August 7th.

When I sync the accounts, actualbudget now shows me two entries for transfer of N euros, one on the 4th August and another entry on 7th of August. This is because the dates for the source and destination of the transfer are not recognized to be linked to the same transaction.

Ok, so I deleted one of the transfer entries and everything is reconciled. But then everytime actualbudget syncs the accounts, a second entry for the transfer appears again in the accounts.

I searched the docs and the issues, and could not find guidance on how to resolve this situation. Ideally I'd like to link the two transactions and file them as a single transfer. This situation that it takes a few days for the transaction to appear on two different bank accounts happens regularly.

Thanks for tips on how to resolve this, and for developing this cool software!

### Where are you hosting Actual?

Fly.io

### What browsers are you seeing the problem on?

Firefox

### Operating System

Linux

**Comment by Jackenmen** [comment]
I am experiencing this kind of problem and I believe that the proper solution would be to support having different transfer dates as requested in #661 - I would suggest upvoting it.

There may be some simpler workarounds allowed to be implemented that would also fix this problem but maintainers will probably have a better idea of whether there's anything that could be done.

**Comment by achennu** [comment]
I had another transfer between accounts that was booked and valued across a weekend. When these were synced through gocardless, it created one linked `transfer` entry between the two accounts. This is shown with a little chain icon in the table of transactions.

 Puzzled, I went back to the older transaction and deleted them in both accounts. After syncing the accounts again, now it turns out that only one transfer entry was made linking the two transactions. So maybe this problem occurred because previously there were two different unique ids being tracked due to staggered syncing schedules -- but I'm only speculating.

I'm using v23.8.1 of the software, in case that matters.

Closing this issue for now.
